test_calculus.py

```diff
@@ -7,114 +7,116 @@
 import pytest
 import numpy as np
 import calculus as calc
-
 # Ctypes initialization routine
 # Load the shared library
 lib_path = os.path.abspath("calculus.dll")
 calculus = ctypes.CDLL(lib_path)
 # Define argument and return types for the DLL functions
 calculus.verify_arguments.argtypes = [ctypes.c_double]
 calculus.verify_arguments.restype = ctypes.c_bool
-
 calculus.calculate_square.argtypes = [ctypes.c_double]
 calculus.calculate_square.restype = ctypes.c_double
-
+lib =ctypes.cdll.LoadLibrary("./calculus.dll")
+lib.adapt_c.argtypes=[ctypes.CFUNCTYPE(ctypes.c_double,ctypes.c_double), ctypes.c_double, ctypes.c_double, ctypes.c_int, ctypes.c_int]
+lib.adapt_c.restype=ctypes.c_double
+def cubic(x):
+    return x**3+1
+def cosine(x):
+    return np.cos(1/x)
+def exponential(x):
+    return np.exp(-1/x)
+    
+cubic_c=ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double)
+python_cubic_c=cubic_c(cubic)
+cos_c=ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double)
+python_cos_c=cos_c(cosine)
+exp_c=ctypes.CFUNCTYPE(ctypes.c_double, ctypes.c_double)
+python_exp_c=exp_c(exponential)
+def test_adapt():
+    """
+    test adapt confirms that the adapt_c function 
+    produces the expected results for certain 
+    integrals.
+    """
+    result1=lib.adapt_c(python_cubic_c, -1,1,10,10)
```

Comment:
I believe the pytest is failing here because above we've imported the functions as calc so this is missing the calc. before adapt_c